doc/dune-files.rst

```diff
@@ -2,10 +2,195 @@
+
+- ``preserve`` does not clear the terminal screen beteween rebuilds.
+
+- ``clear-on-rebuild`` clears the terminal screen between rebuilds.
+
+- ``clear-on-rebuild-and-flush-history`` clears the terminal between rebuilds, and
+  it also deletes everything in the scrollback buffer.
+
+.. _sandboxing_preference:
+
+``sandboxing_preference``
+-------------------------
+
+The preferred sandboxing setting. Individual rules may specify different
+preferences. Dune will try to utilize a setting satisfying both conditions.
+
+.. code:: scheme
+
+    (sandboxing_preference <setting> <setting> ...)
+
+where each ``<setting>`` can be one of:
+
+- ``none``, no sandboxing allowed
+
+- ``symlink``, using symbolic links
+
+- ``copy``, using file copies
+
+- ``hardlink``, using hard links
+
+.. _cache:
+
+``cache``
+---------
+
+Specifies whether to utilize the Dune cache.
+
+.. code:: scheme
+
+    (cache <setting>)
+
+where ``<setting>`` is one of:
+
+- ``enabled`` enables Dune cache.
+
+- ``disabled`` disables Dune cache.
+
+.. _cache_check_probability:
+
+``cache-check-probability``
+---------------------------
+
+While the main purpose of Dune cache is to speed up build times, it can also be
+used to check build reproducibility. It is possible to enable a probabilistic
+check, in which Dune will re-execute randomly chosen build rules and compare
+their results with those stored in the cache. If the results differ, the rule is
+not reproducible, and Dune will print out a corresponding warning.
+
+.. code:: scheme
+
+    (cache-check-probability <number>)
+
+where ``<number>`` is a floating-point number between 0 and 1 (inclusive). 0
+means never to check for reproducibility, and 1 means to always perform the
+check.
+
+.. _cache_storage_mode:
+
+``cache-storage-mode``
+----------------------
+
+Specify the mechanism used by the Dune cache storage layer.
+
+.. code:: scheme
+
+    (cache-storage-mode <setting>)
+
+where ``<setting>`` is one of:
+
+- ``auto``, Dune decides the best mechanism to use.
+
+- ``hardlink``, using hard links.
+
+- ``copy``, using file copies; this is less efficient than using hard links.
+
+.. _action_stdout_on_success:
+
+``action_stdout_on_success``
+----------------------------
+
+Specifies how Dune should handle the standard output of actions when they succeed.
+This can be used to reduce the noise of large builds.
+
+.. code:: scheme
+
+    (action_stdout_on_success <setting>)
+
+where ``<setting>`` is one of:
+
+- ``print`` prints the output on the terminal (this is the default).
+
+- ``swallow`` ignores the output and does not print it on the terminal.
+
+- ``must-be-empty`` enforces that the output should be empty. If it is not, Dune will fail.
+
+.. _action_stderr_on_success:
+
+``action_stderr_on_success``
+----------------------------
+
+Same as :ref:`action_stdout_on_success`, but applies to standard error instead
+of standard output.
+
 .. _dune-project:
 
-dune-project
-============
+``dune-project``
```

Comment:
Let's not do this as it makes the headers weird.

Reply:
Fixed.